projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DataJobsSynchronizer.java

```diff
@@ -59,25 +70,48 @@ public DataJobsSynchronizer(
    */
   public void synchronizeDataJobs() throws ApiException {
     ThreadPoolTaskExecutor taskExecutor = initializeTaskExecutor();
-    Set<String> cronJobs = dataJobsKubernetesService.listCronJobs();
+    Set<String> dataJobDeploymentNamesFromKubernetes = dataJobsKubernetesService.listCronJobs();
+    Iterable<DataJob> dataJobsFromDB = jobsRepository.findAll();
+
+    Map<String, DesiredDataJobDeployment> desiredDataJobDeploymentsFromDBMap =
+        desiredJobDeploymentRepository.findAll().stream()
+            .collect(
+                Collectors.toMap(DesiredDataJobDeployment::getDataJobName, Function.identity()));
+
+    Map<String, ActualDataJobDeployment> actualDataJobDeploymentsFromDBMap =
+        actualJobDeploymentRepository.findAll().stream()
+            .collect(
+                Collectors.toMap(ActualDataJobDeployment::getDataJobName, Function.identity()));
 
-    jobsRepository
-        .findAll(
-            Sort.by(
-                Sort.Direction.ASC,
-                DataJob_.DATA_JOB_DEPLOYMENT + "." + DataJobDeployment_.DEPLOYMENT_VERSION_SHA))
-        .forEach(dataJob -> taskExecutor.execute(() -> synchronizeDataJob(dataJob, cronJobs)));
+    dataJobsFromDB.forEach(
+        dataJob ->
+            taskExecutor.execute(
+                () ->
+                    synchronizeDataJob(
+                        dataJob,
+                        desiredDataJobDeploymentsFromDBMap.get(dataJob.getName()),
+                        actualDataJobDeploymentsFromDBMap.get(dataJob.getName()),
+                        dataJobDeploymentNamesFromKubernetes.contains(dataJob.getName()))));
 
     taskExecutor.shutdown();
```

Comment:
Btw. I am not sure shutdown guarantees teh tasks are completed. There is taskExecutor.awaitTermination for that.

Reply:
It will wait because of executor.setWaitForTasksToCompleteOnShutdown(true); but probably we will change the async approach based on your comment in one of the previous PRs.